linalg/hypre.cpp

```diff
@@ -29,15 +29,28 @@ namespace mfem
 
 bool Hypre::configure_runtime_policy_from_mfem = true;
 
-Hypre::Hypre()
+void Hypre::Init()
 {
-#if MFEM_HYPRE_VERSION >= 21900
-   // Initializing hypre
-   HYPRE_Init();
+   Hypre &hypre = Instance();
+   if (hypre.state == State::NONE)
```

Comment:
It may be a good idea to allow hypre initialization after hypre is finalized as well:
```suggestion
   if (hypre.state != State::INITIALIZED)
```